### PR TITLE
Update Recipe Schema and RecipeAPI

### PR DESCRIPTION
**Pull request by hwbit**
Schema update includes visibility and tags.
Recipe API has error handling logic and bug fix.

### CODE REVIEW

File: preppal-be/models/recipe.ts [hunk 1]
```diff
@@ -50,6 +50,12 @@ const RecipeSchema = new Schema({
     },
     modifiedDate: {
         type: Date
+    },
+    tags: {
+        type: Array
+    },
+    visibility: {
+        type: String
```
Comment (algorizan) [inline preppal-be/models/recipe.ts:58]: Why String and not a simple boolean flag like true for public, false for private?

File: preppal-be/models/recipe.ts [hunk 2]
```diff
@@ -50,6 +50,12 @@ const RecipeSchema = new Schema({
     },
     modifiedDate: {
         type: Date
+    },
+    tags: {
+        type: Array
```
Comment (algorizan) [inline preppal-be/models/recipe.ts:55]: Is there a way to specify the type of the items in the Array as well?

File: preppal-be/models/recipe.ts (hunk 1 above)
Comment (hwbit) [reply preppal-be/models/recipe.ts]: fixed

File: preppal-be/models/recipe.ts (hunk 2 above)
Comment (hwbit) [reply preppal-be/models/recipe.ts]: fixed.

File: preppal-be/routes/recipeApi.ts [hunk 3]
```diff
@@ -90,10 +97,21 @@ routerRecipeApi.post("/createRecipe", async (req, res) => {
...
     try {
-        const { _id, author, title, description, image, ingredients, instructions, servingSize, prepTime, cookingTime, creationDate } = req.body;
+        const { _id, author, title, description, image, ingredients, instructions, servingSize, prepTime, cookingTime, creationDate, tags, visibility } = req.body;
+        
+        const verifyRecipe = await Recipe.findOne({ _id: _id });
+
+        if (!verifyRecipe) {
+            return res.status(400).json({ errors: [{ msg: "Invalid Id for recipe."}] });
+        } else if (!author) {
+            return res.status(404).json({ msg: "Recipe requires an author." });
```
Comment (lik5) [inline preppal-be/routes/recipeApi.ts:107]: This should probably be a status 400 code
I think 404 means not found and 400 means bad request

File: preppal-be/routes/recipeApi.ts [hunk 4]
```diff
@@ -90,10 +97,21 @@ routerRecipeApi.post("/createRecipe", async (req, res) => {
...
+        } else if (!title) {
+            return res.status(404).json({ msg: "Recipe requires a title." });
```
Comment (lik5) [inline preppal-be/routes/recipeApi.ts:109]: This should probably be a status 400 code
I think 404 means not found and 400 means bad request

**Review by lik5 — COMMENTED** [review]
Just need the same update on the status codes I think

**Review by lik5 — APPROVED** [review]

**Pull request merged**